metal/config.go

```diff
@@ -49,14 +87,7 @@ func MetalRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool
 
 // Client returns a new client for accessing Equinix Metal's API.
 func (c *Config) Client() *packngo.Client {
-	httpClient := retryablehttp.NewClient()
-	httpClient.RetryWaitMin = time.Second
-	httpClient.RetryWaitMax = 30 * time.Second
-	httpClient.RetryMax = 10
-	httpClient.CheckRetry = MetalRetryPolicy
-	httpClient.HTTPClient.Transport = logging.NewTransport(
-		"Equinix Metal",
-		httpClient.HTTPClient.Transport)
-
+	transport := logging.NewTransport("Equinix Metal", http.DefaultTransport)
+	httpClient := &http.Client{Transport: newRetryableTransport(transport)}
```

Comment:
Is there any advantage of these client changes over what was proposed in https://github.com/packethost/packngo/pull/222#issue-533950813 (with less change)?

Reply:
```
retryClient := retryablehttp.NewClient()
retryClient.RetryMax = 10
retryClient.RetryWaitMin = time.Second
retryClient.RetryWaitMax = 30 * time.Second
retryClient.RetryMax = 10
retryClient.CheckRetry = packngo.RetryPolicy
httpClient := retryClient.StandardClient()
c := NewClientWithAuth(consumerToken, apiToken, httpClient)
```

Reply:
I think it should be `httpClient.HTTPClient.Transport`.

https://github.com/hashicorp/go-retryablehttp/blob/991b9d0a42d13014e3689dd49a94c02be01f4237/client.go#L772

Reply:
As a separate issue, the RetryMax and RetryWaitMax constants should be promoted to provider arguments and/or environment variables rather than being hardcoded here.

RetryMax -> max_retries? https://registry.terraform.io/providers/hashicorp/aws/latest/docs#max_retries
RetryWaitMax -> max_retry_wait?

Reply:
@displague thanks for the review, yeah you're right, your code is simpler and works. I fixed it.

I also added the retry params to the provider.

Reply:
.. but I think that pulling the retry policy from packngo is not a good idea, the provider should define its own (just like any other tool using packngo as a dep).

Reply:
The old PR comment suggested the packngo RetryPolicy. The code snippet above continues to use the MetalRetryPolicy.

Reply:
(You may be seeing an older copy of my edited comment)